File: Backend/Models/Events/EventType.cs
namespace Helse.Models.Events;

public class CreateEventType : BaseEventType
{
}

public class UpdateEventType : CreateEventType
{
    public required long Id { get; set; }
}

public class EventType : BaseEventType
{
    public required long Id { get; set; }

    public required bool UserEditable { get; set; }
}

public abstract class BaseEventType
{
    public required string Name { get; set; }

    public string? Description { get; set; }

    public bool StandAlone { get; set; }

    public bool Visible { get; set; }

    public TimeSpan? TimeDifference { get; set; }

    public required long GroupId { get; set; }
}

File: flutter_frontend/lib/services/local/database/models/event_type.dart
import 'package:drift/drift.dart';
import 'package:helse/services/local/database/models/helse_table.dart';
import 'package:helse/services/local/database/models/metric.dart';

class EventType extends HelseTable {
  TextColumn get name => text()();
  TextColumn get description => text().nullable()();
  BoolColumn get standAlone => boolean()();
  BoolColumn get visible => boolean()();
  TextColumn get timeDifference => text().nullable()();
  BoolColumn get userEditable => boolean()();
  late final groupId = integer().references(Group, #id)();
}
